### PR TITLE
Switch from Travis CI to GitHub Actions

### PR DESCRIPTION
**Pull request by devtonhere**

### CODE REVIEW

File: .github/scripts/deploy.sh [hunk 1]
```diff
@@ -0,0 +1,9 @@
+#!/bin/bash
+set -ev
+
+# Import gpg key
+echo $GPG_PRIVATE_KEY > private.key
+gpg --fast-import private.key
+
+# Deploy to Maven Central
+mvn --settings .github/travis-settings.xml clean deploy -Prelease
```
Comment (dmitriy-abramov) [inline .github/scripts/deploy.sh:9]: You have renamed `travis-settings.xml` the script won't work.

**Review by dmitriy-abramov — CHANGES_REQUESTED** [review]

File: .github/scripts/deploy.sh (hunk 1 above)
Comment (devtonhere) [reply .github/scripts/deploy.sh]: Fixed

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -0,0 +1,23 @@
+name: Test
+
+on:
+  pull_request:
+    types: [ synchronize ]
```
Comment (devtonhere) [inline .github/workflows/test.yml:5]: verify if this is correct (do not trigger workflow when PR is closed)

File: .github/workflows/test.yml [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+jobs:
+  build:
```
Comment (devtonhere) [inline .github/workflows/test.yml:8]: rename to test

File: .github/workflows/release.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+      run: .github/scripts/test.sh
+    - name: Push git tag
+      run: .github/scripts/push-tag.sh
+    - name: Deploy Release
+      env:
+        GPG_KEY_NAME: ${{ secrets.GPG_KEY_NAME }}
+        GPG_PASSPHRASE: ${{ secrets.GPG_PASSPHRASE }}
+        OSSRH_USERNAME: ${{ secrets.OSSRH_USERNAME }}
+        OSSRH_PASSWORD: ${{ secrets.OSSRH_PASSWORD }}
+        GPG_PRIVATE_KEY: ${{ secrets.GPG_PRIVATE_KEY }}
```
Comment (devtonhere) [inline .github/workflows/release.yml:33]: Prevent accidentally printing any secrets.

Comment (devtonhere) [reply .github/workflows/release.yml:33]: According to this comment it's built-in: https://stackoverflow.com/a/59482124

**Review by dmitriy-abramov — APPROVED** [review]

**Pull request merged**